library/core/src/main/java/com/google/android/exoplayer2/source/SilenceMediaSource.java

```diff
@@ -33,6 +33,30 @@
 /** Media source with a single period consisting of silent raw audio of a given duration. */
 public final class SilenceMediaSource extends BaseMediaSource {
 
+  public static final class Factory {
+
+    @Nullable private Object tag;
+    private long duration;
+    private boolean isCreateCalled;
+
+    public SilenceMediaSource.Factory setTag(Object tag) {
+      Assertions.checkState(!isCreateCalled);
```

Comment:
We made factories of other media sources re-usable. So create can be called multiple times.  I'd probably remove this for consistency.

Reply:
I just followed how the ProgressiveMediaSource.Factory is implemented. So should I remove the boolean and the assertion ?